libs/SalesforceSDK/src/com/salesforce/androidsdk/auth/idp/IDPCodeGeneratorActivity.java

```diff
@@ -121,6 +123,14 @@ public boolean shouldOverrideUrlLoading(WebView view, String url) {
                 } else {
                     handleError("Code not returned from server");
                 }
+            } else if (url.contains(EC_301) || url.contains(EC_302)) {
```

Comment:
This one's a bit ugly, but I see no other way until we get an API that's truly a first-class citizen. @trooper2013 You should probably add this error handling on the `iOS` side too.